src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -315,91 +429,307 @@ private JPanel getPanelContent() {
+						inventory.setDestination(null);
+					}
+					inventory = medicalInventoryManager.updateMedicalInventory(inventory);
+					for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator.hasNext();) {
+						MedicalInventoryRow medicalInventoryRow = iterator.next();
+						Medical medical = medicalInventoryRow.getMedical();
+						Lot lot = medicalInventoryRow.getLot();
+						String lotCode ;
+						medicalInventoryRow.setInventory(inventory);
+						if (medicalInventoryRow.getId() == 0) {
+							if (lot != null) {
+								lotCode = lot.getCode();
+								boolean isExist = false;
+								Lot lotExist = movStockInsertingManager.getLot(lotCode);
+								if (lotExist != null) {
+									isExist = true;
+								}
+								if (!isExist) {
+									if (lot.getDueDate() != null) {
+										Lot lotStore = movStockInsertingManager.storeLot(lotCode, lot, medical);
+										medicalInventoryRow.setLot(lotStore);
+										medicalInventoryRow.setNewLot(true);
+									} else {
+										medicalInventoryRow.setLot(null);
+									}
+								} else {
+									Lot lotStore = movStockInsertingManager.updateLot(lot);
+									medicalInventoryRow.setLot(lotStore);
+								}
+							}
+							medicalInventoryRowManager.newMedicalInventoryRow(medicalInventoryRow);
+						} else {
+							lot = medicalInventoryRow.getLot();
+							if (lot != null) {
+								lotCode = lot.getCode();
+								Lot lotExist = movStockInsertingManager.getLot(lotCode);
+								if (lotExist != null) {
+									Lot lotStore;
+									lotExist.setDueDate(lot.getDueDate());
+									lotExist.setPreparationDate(lot.getPreparationDate());
+									lotExist.setCost(lot.getCost());
+									lotStore = movStockInsertingManager.updateLot(lotExist);
+									medicalInventoryRow.setLot(lotStore);
+								} else {
+									if (lot.getDueDate() != null) {
+										Lot lotStore = movStockInsertingManager.storeLot(lotCode, lot, medical);
+										medicalInventoryRow.setLot(lotStore);
+										medicalInventoryRow.setNewLot(true);
+									} else {
+										medicalInventoryRow.setLot(null);
+									}
+								}
+							} else {
+								medicalInventoryRow.setLot(null);
+							}
+							if (medicalInventoryRow.getId() == 0) {
+								medicalInventoryRowManager.newMedicalInventoryRow(medicalInventoryRow);
+							} else {
+								medicalInventoryRowManager.updateMedicalInventoryRow(medicalInventoryRow);
+							}
+						}
+					}
+					MessageDialog.info(null, "angal.inventory.update.success.msg");
+					resetVariable();
+					fireInventoryUpdated();
+					int info = MessageDialog.yesNo(null, "angal.inventoryrow.doyouwanttocontinueediting.msg");
+					if (info != JOptionPane.YES_OPTION) {
+						dispose() ; 
 					}
-				} catch (OHServiceException e) {
-					OHServiceExceptionUtil.showMessages(e);
 				}
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+				return ;
```

Comment:
Please replace all " ;" with ";"

```suggestion
				return;
```